### DISCUSSION TITLE
Problem with waterfox-g4-kde offline package (missing packages)

### DISCUSSION THREAD

**Discussion by 4evermaat** [hawkeye116477/waterfox-deb-rpm-arch-AppImage#43] [General]
I was following the instructions here as of 06 Jan 2022.
https://software.opensuse.org//download.html?project=home%3Ahawkeye116477%3Awaterfox&package=waterfox-g4-kde
under the `Grab binary packages directly` `>` `Packages for xUbuntu 20.04:` section, which points to this link.

https://download.opensuse.org/repositories/home:/hawkeye116477:/waterfox/xUbuntu_20.04/amd64/waterfox-g4-kde_0.5.1-0+56.1_amd64.deb

When I download deb file, it is 2.4Kb file large.  and I keep getting error `unable to install ......the following package has unmet dependencies` 

Can someone please correct the .deb binaries so that they contain the full offline install package materials?

**Comment by hawkeye116477** [comment]
See Note 1 on readme, same is also true for waterfox-g4-kde, that package installs two other packages available on repository and suggests one (https://github.com/hawkeye116477/waterfox-deb-rpm-arch-AppImage/blob/master/waterfox-g4-kpe/control#L26). It's blank, cuz the only it's purpose is to make life easier for KDE users, who needs additional depends for KDE integration (unfortunately deb packages hasn't `Supplements` feature, so that's why it's done that way).

So to sum up for KDE you need: `waterfox-g4-kpe`, `kwaterfoxhelper` and `optionally plasma-browser-integration from distro's repo (+extension from AMO)`.

**Comment by 4evermaat** [comment]
So are you suggesting that I use the command line option instead? will that auto-grab all necessary dependencies?

I'm new to Ubuntu and honestly your empty deb package is very deceptive.  I've downloaded other apps deb and everything is there. 

Can you design a deb that assumes the user wants a 1-2 click option to get Waterfox on their machine.  Either installing for the first time....or reinstalling/upgrading from a previous install.

Forgive me if I'm still thinking from a Windows/Mac gui mindset.  Although other ubuntu deb packages were not so difficult.

Or should I just use the "portable" generic linux version [from the official website](https://github.com/WaterfoxCo/Waterfox/releases/tag/G4.0.5.1)?.  and if so, what are the pros/cons of using portable version?

**Comment by 4evermaat** [comment]
ok.  So after trying alternate ways to install deb with no success, the [`add repository and install manually`](https://software.opensuse.org//download.html?project=home%3Ahawkeye116477%3Awaterfox&package=waterfox-g4-kde) option works flawlessly.  Specifically I used this one:

`For xUbuntu 20.04 run the following:`
<p>

Each line as separate command in `terminal`

```
echo 'deb http://download.opensuse.org/repositories/home:/hawkeye116477:/waterfox/xUbuntu_20.04/ /' | sudo tee /etc/apt/sources.list.d/home:hawkeye116477:waterfox.list
curl -fsSL https://download.opensuse.org/repositories/home:hawkeye116477:waterfox/xUbuntu_20.04/Release.key | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/home_hawkeye116477_waterfox.gpg > /dev/null
sudo apt update
sudo apt install waterfox-g4-kde
```
---

I still argue that grab binary packages directly (`.deb` files) need improvement.

**Comment by hawkeye116477** [comment]
> I still argue that grab binary packages directly (.deb files) need improvement.
> Can you design a deb that assumes the user wants a 1-2 click option to get Waterfox on their machine. Either installing for the first time....or reinstalling/upgrading from a previous install.

Well, for non-KDE users there is `waterfox-g4-kpe` package available, which doesn't requires depends from my repository.
Unfortunately if some package needs something from my repository, then installing from offline package will always cause problems.

In that case if you wanted to install `waterfox-g4-kde` with offline package, then you needed to do by manual job which does that package, so download and install [waterfox-g4-kpe](https://software.opensuse.org//download.html?project=home%3Ahawkeye116477%3Awaterfox&package=waterfox-g4-kpe) and [kwaterfoxhelper](https://software.opensuse.org//download.html?project=home%3Ahawkeye116477%3Awaterfox&package=kwaterfoxhelper) in that order and possibly **plasma-browser-integration** from Ubuntu repo and [Plasma Integration extension](https://addons.mozilla.org/pl/firefox/addon/plasma-integration/).
